resources/js/contest-judge/entry.tsx

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import BigButton from 'components/big-button';
+import ContestEntryJson from 'interfaces/contest-entry-json';
+import ContestScoringCategoryJson from 'interfaces/contest-scoring-category-json';
+import { route } from 'laroute';
+import { action, computed, makeObservable, observable, runInAction } from 'mobx';
+import { observer } from 'mobx-react';
+import { ContestEntry } from 'models/contest-entry';
+import * as React from 'react';
+import ContestJudgeStore from 'stores/contest-judge-store';
+import { onError } from 'utils/ajax';
+import { trans } from 'utils/lang';
+import { CurrentUserJudgeVote } from './current-user-judge-vote';
+
+interface Props {
+  entry: ContestEntry;
+  store: ContestJudgeStore;
+}
+
+@observer
+export default class Entry extends React.Component<Props> {
+  @observable private readonly currentVote = new CurrentUserJudgeVote();
+  @observable private readonly initialVote = new CurrentUserJudgeVote();
+  @observable private posting = false;
+  @observable private xhr?: JQuery.jqXHR;
+
+  constructor(props: Props) {
+    super(props);
+
+    const voteFromJson = props.entry.current_user_judge_vote;
+    if (voteFromJson != null) {
+      this.currentVote.updateWithJson(voteFromJson);
+      this.initialVote.updateWithJson(voteFromJson);
+    }
+
+    makeObservable(this);
+  }
+
+  @computed
+  private get disabled() {
+    const scoresHaveChanged = this.props.store.scoringCategories.some((category) => {
+      const score = this.currentVote.scores.get(category.id);
+      if (score == null) return false;
+
+      const initialScore = this.initialVote.scores.get(category.id);
+      if (initialScore?.value !== score.value) return true;
+
+      return false;
+    });
+
+    if (
+      !scoresHaveChanged &&
+      this.currentVote.comment === this.initialVote.comment
+    ) return true;
+
+    return false;
```

Comment:
Also should check if all the scores are set (I'm assuming the comment is not required here?)

```suggestion
    return !(
      this.currentVote.scores.size === this.props.store.scoringCategories.length
        && (scoresHaveChanged || this.currentVote.comment !== this.initialVote.comment)
    );
```

Reply:
the comment is not required yes